src/simpledb/parse/Parser.java

```diff
@@ -81,22 +81,29 @@ public Predicate predicate() {
       }
       return pred;
    }
-   
-// Methods for parsing queries
-   
+
+   // Methods for parsing queries
+
    public QueryData query() {
       lex.eatKeyword("select");
       List<String> fields = selectList();
       lex.eatKeyword("from");
       Collection<String> tables = tableList();
       Predicate pred = new Predicate();
+      List<String> sortfields = new ArrayList<>();
       if (lex.matchKeyword("where")) {
          lex.eatKeyword("where");
          pred = predicate();
       }
-      return new QueryData(fields, tables, pred);
+      if (lex.matchKeyword("order")) {
+         lex.eatKeyword("order");
+         lex.eatKeyword("by");
+         // TODO: handle optional 'asc' and 'desc' keywords
+         sortfields = selectList();

```

Comment:
I foresee reimplementing this as a list of sortfields instead, to handle an optional `asc/desc` keyword.

For this, I intend to:
1. Create a `SortField` class comprising of the attributes `field` and `order`(default to `asc`)
2. Modify the above code to take a `List<SortField>` and a helper method to eat the keywords (similar to `selectList()`)
3. Modify `QueryData` class to hold the attribute `List<SortField> sortfields`